platform/search/src/SearchAggregator.js

```diff
@@ -0,0 +1,144 @@
+/*****************************************************************************
+ * Open MCT Web, Copyright (c) 2014-2015, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+
+/**
+ * Module defining SearchAggregator. Created by shale on 07/16/2015.
+ */
+define(
+    [],
+    function () {
+        "use strict";
+
+        var DEFUALT_TIMEOUT =  1000,
+            DEFAULT_MAX_RESULTS = 100;
+        
+        /**
+         * Allows multiple services which provide search functionality 
+         *   to be treated as one.
+         *
+         * @constructor
+         * @param $q Angular's $q, for promise consolidation.
+         * @param {SearchProvider[]} providers The search providers to be
+         *        aggregated.
+         */
+        function SearchAggregator($q, providers) {
+            
+            // Remove duplicate objects that have the same ID. Modifies the passed 
+            //   array, and returns the number that were removed. 
+            function filterDuplicates(results, total) {
+                var ids = [],
```

Comment:
I'd use `ids = {}` here instead of an array, and replace the `indexOf` check I'd do an `ids[results[i].id]`. Finally, instead of pushing into `ids`, you'd do `ids[results[i].id] = true`

The reasoning for this is that `Array.prototype.indexOf` check is going to be O(n), whereas looking up a value from an object should be amortized O(1); since this is done in done in a loop through all results, `filterDuplicates` as a whole would be O(n<sup>2</sup>) without this change, and O(n) with it. (Where n is the total number of results.)